### PR TITLE
Fixed bug in DynamicDataService.java eviction logic.

### PR DESCRIPTION
**Pull request by a-badaev**

### CODE REVIEW

**Comment by anton-antonenko** [comment]
:+1:

File: jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java [hunk 1]
```diff
@@ -38,22 +37,25 @@
     
     private static final Logger LOGGER = LoggerFactory.getLogger(DynamicDataService.class);
     
-    private final ExecutorService destroyerService =
-            Executors.newSingleThreadExecutor(r -> new Thread("DynamicDataServiceDestoyer"));
+    private final ExecutorService destroyerService = Executors.newSingleThreadExecutor(r -> {
```
Comment (SokolAndrey) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java]: Why don't you use new Thread(r, "DestroyerServiceThread")?

File: jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java [hunk 2]
```diff
@@ -66,11 +68,29 @@ public void destroy() {
         destroyerService.shutdown();
     }
     
-    protected void evictDataServiceFor(final Long configId) {
-        LOGGER.info("Evicting jagger test db config with id: {}", configId);
+    protected void evict(final Long configId) {
+        destroyerService.execute(() -> doEvict(configId));
+    }
+    
+    protected void evictIfAboveThreshold() {
```
Comment (SokolAndrey) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java]: It seems quite strange to me. Maybe there is any possibility to make it in a simple way?

File: jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java (hunk 1 above)
Comment (a-badaev) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java]: Yes. I can. But it is even better to extend existing default implementation, isn't it?

File: jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java (hunk 2 above)
Comment (a-badaev) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java]: It looks pretty simple to me. There are just 3 lines. Any proposals how to simplify?

File: jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java [hunk 3]
```diff
@@ -84,14 +104,8 @@ public DataService getDataServiceFor(final Long configId) {
...
-            synchronized (this) {
-                dataServiceContext = dataServiceContexts.computeIfAbsent(configId, s -> {
-                    if (dataServiceContexts.size() >= dataServiceCacheSize) {
-                        evictDataServiceFor(dataServiceContexts.keySet().iterator().next());
-                    }
-                    return initDataServiceContextFor(config);
-                });
-            }
+            evictIfAboveThreshold();
+            dataServiceContext = dataServiceContexts.computeIfAbsent(configId, s -> initDataServiceContextFor(config));
```
Comment (SokolAndrey) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java]: The _s_ parameter seems to be useless.

File: jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java (hunk 1 above)
Comment (SokolAndrey) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java]: I consider we should stick to KISS principle. So, (r -> new Thread(Runnable r, String name)) seems to be less complicated.

File: jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java (hunk 3 above)
Comment (a-badaev) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java]: you can't omit it.

File: jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java [hunk 4]
```diff
@@ -37,11 +37,7 @@
     
     private static final Logger LOGGER = LoggerFactory.getLogger(DynamicDataService.class);
     
-    private final ExecutorService destroyerService = Executors.newSingleThreadExecutor(r -> {
-        Thread thread = Executors.defaultThreadFactory().newThread(r);
-        thread.setName("DestroyerServiceThread");
-        return thread;
-    });
+    private final ExecutorService destroyerService = Executors.newSingleThreadExecutor(r -> new Thread(r, "DynamicDataServiceDestoyer"));
```
Comment (a-badaev) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java]: fixed

**Pull request merged**